setup.py

```diff
@@ -227,7 +227,7 @@ def run(self) -> None:
 
 
 setup_requirements = [
-    "setuptools",
+    "setuptools>=64.0.0",
```

Comment:
I'm not sure if we want to straight up require a version of setuptools this recent, this was just released on Aug 11 - what doesn't work on older versions?  You can `pip install .`, but only doing a clean build?  And you can't `pip install -e .` at all?  We were definitely capable of doing `pip install -e .` before last month...  We could also not mark this as required in `setup.py` but include notes on these in the "Building from source" section of the readme?

Reply:
Well, the `pip install -e .` just puts in the install in the wrong place for older versions. Never did finish the investigation into why and how to get it put in correct place. I didn't see any harm to having users upgrade their setuptools verisons. Including a note in the readme could work, but I was just thinking it would nice to have it just work.

Reply:
> Well, the pip install -e . just puts in the install in the wrong place for older versions. Never did finish the investigation into why and how to get it put in correct place

I remain bamboozled by this, but I agree I'm ok not investigating it since we have other methods of installation on those versions of setuptools.  I'd much rather mention in the README at the top of each installation method if a particular minimum version of setuptools is required, or a clean build directory is required, rather than refusing to install outright

> I didn't see any harm to having users upgrade their setuptools verisons. Including a note in the readme could work, but I was just thinking it would nice to have it just work.

The harm would be that it isn't always easy to just upgrade your setuptools version for a variety of reasons - for instance other packages will have their setuptools versions pinned for the same reasons we pin our python packages, other packages may not support the bleeding edge version of setuptools yet and may break entirely, and some users may just not be able to update their setuptools to latest, e.g. if they're using system-python-installed setuptools and their distro package manager doesn't have latest setuptools